planning/behavior_velocity_planner/test/src/test_grid_utils.cpp

```diff
@@ -101,3 +104,113 @@ TEST(isOcclusionSpotSquare, occlusion_spot_cell)
+using behavior_velocity_planner::Point2d;
+using behavior_velocity_planner::Polygon2d;
+using behavior_velocity_planner::grid_utils::occlusion_cost_value::OCCUPIED;
+using behavior_velocity_planner::grid_utils::occlusion_cost_value::UNKNOWN;
+
+Polygon2d pointsToPoly(const Point2d p0, const Point2d p1, const double radius)
+{
+  LineString2d line = {p0, p1};
+  const double angle = atan2(p0.y() - p1.y(), p0.x() - p1.x());
+  const double r = radius;
+  Polygon2d line_poly;
+  // add polygon counter clockwise
+  line_poly.outer().emplace_back(p0.x() + r * sin(angle), p0.y() - r * cos(angle));
+  line_poly.outer().emplace_back(p1.x() + r * sin(angle), p1.y() - r * cos(angle));
+  line_poly.outer().emplace_back(p1.x() - r * sin(angle), p1.y() + r * cos(angle));
+  line_poly.outer().emplace_back(p0.x() - r * sin(angle), p0.y() + r * cos(angle));
+  // std::cout << boost::geometry::wkt(line_poly) << std::endl;
+  // std::cout << boost::geometry::wkt(line) << std::endl;
+  return line_poly;
+}
+
+TEST(compareTime, polygon_vs_line_iterator)
+{
+  // Prepare an occupancy grid with a single occlusion_spot
+
+  /*
+      0 1 2 ....200
+    0
+    1   ? ? ?    ?
+    2   ? ? ?    .
+    .   ? ? ?    ?
+    200 ? ? ? ...?
+    */
+  const size_t cell_size = 200;
+  grid_map::GridMap grid = test::generateGrid(cell_size, cell_size, 0.5);
+  std::vector<grid_map::Index> unknown_cells(cell_size);
+  for (int i = 0; i < grid.getLength().x(); ++i) {
+    for (int j = 0; j < grid.getLength().y(); ++j) {
+      const grid_map::Index index = {i, j};
+      grid.at("layer", index) = UNKNOWN;
+    }
+  }
+  const grid_map::Matrix & grid_data = grid["layer"];
+  tier4_autoware_utils::StopWatch<std::chrono::milliseconds> stop_watch;
+  stop_watch.tic("processing_time");
+  size_t count = 0;
+  for (size_t i = 0; i < 10; i++) {
+    const double length = static_cast<double>(cell_size * i) * 0.005 * std::sqrt(2);
+    std::cout << "length of polygon: " << length << std::endl;
+    // polygon iterator
+    {
+      const Point2d p0(0, 0);
+      const Point2d p1(length, length);
+      const Polygon2d polygon = pointsToPoly(p0, p1, 0.5);
+      grid_map::Polygon grid_polygon;
+      try {
+        for (const auto & point : polygon.outer()) {
+          grid_polygon.addVertex({point.x(), point.y()});
+        }
+        for (grid_map::PolygonIterator iterator(grid, grid_polygon); !iterator.isPastEnd();
+             ++iterator) {
+          const grid_map::Index & index = *iterator;
+          if (grid_data(index.x(), index.y()) == OCCUPIED) {
+            std::cout << "occupied" << std::endl;
+            // occupied
+          } else {
+            count++;
+            // continue
+          }
+        }
+      } catch (const std::invalid_argument & e) {
+        std::cerr << e.what() << std::endl;
+      }
+    }
+    std::cout << "polygon iterator [ms]: " << stop_watch.toc("processing_time", true)
+              << " :num: " << count << std::endl;
```

Comment:
pls remove

Reply:
@yukkysaito 
I make it silent at [761e310](https://github.com/autowarefoundation/autoware.universe/pull/568/commits/761e31039efd8fe26e133e9317c1fa9e3537fb8b)